custom/icds_reports/management/commands/fill_bihar_api_data.py

```diff
@@ -0,0 +1,96 @@
+STATE_ID = 'f9b47ea2ee2d8a02acddeeb491d3e175'
+
+
+def _run_custom_sql_script(command, day=None):
+    db_alias = get_icds_ucr_citus_db_alias()
+    if not db_alias:
+        return
+    with connections[db_alias].cursor() as cursor:
+        cursor.execute(command, [day])
+
+
+class Command(BaseCommand):
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            'start_date',
+            type=date_type,
+            help='The start date (inclusive). format YYYY-MM-DD',
+            nargs='?'
+        )
+
+    def build_bp_data(self, month):
+        if month == '2020-03-01':
+            path = os.path.join(os.path.dirname(__file__), 'sql_scripts', 'build_bp_initial_data.sql')
+            with open(path, "r", encoding='utf-8') as sql_file:
+                sql_to_execute = sql_file.read()
+                sql_to_execute = sql_to_execute.format(month=month.strftime("%Y-%m-%d"), state_id=STATE_ID)
+                _run_custom_sql_script(sql_to_execute)
+        else:
+            icds_state_aggregation_task(state_id=STATE_ID, date=month,
+                                        func_name='_aggregate_bp_forms')
+
+    def build_pnc_data(self, month):
+        if month == '2020-03-01':
+            path = os.path.join(os.path.dirname(__file__), 'sql_scripts', 'build_pnc_initial_data.sql')
+            with open(path, "r", encoding='utf-8') as sql_file:
+                sql_to_execute = sql_file.read()
+                sql_to_execute = sql_to_execute.format(month=month.strftime("%Y-%m-%d"), state_id=STATE_ID)
+                _run_custom_sql_script(sql_to_execute)
+        else:
+            icds_state_aggregation_task(state_id=STATE_ID, date=month,
+                                        func_name='_aggregate_ccs_record_pnc_forms')
+
+    def build_migration_forms(self, month):
+        if month == '2020-03-01':
+            path = os.path.join(os.path.dirname(__file__), 'sql_scripts', 'build_migration_initial_data.sql')
+            with open(path, "r", encoding='utf-8') as sql_file:
+                sql_to_execute = sql_file.read()
+                sql_to_execute = sql_to_execute.format(month=month.strftime("%Y-%m-%d"), state_id=STATE_ID)
+                _run_custom_sql_script(sql_to_execute)
+        else:
+            icds_state_aggregation_task(state_id=STATE_ID, date=month,
+                                        func_name='_agg_migration_table')
+
+    def update_ccs_data(self, month):
+        path = os.path.join(os.path.dirname(__file__), 'sql_scripts', 'update_ccs_data.sql')
+        with open(path, "r", encoding='utf-8') as sql_file:
+            sql_to_execute = sql_file.read()
+            sql_to_execute = sql_to_execute.format(month=month.strftime("%Y-%m-%d"), state_id=STATE_ID)
+            sql_to_execute = sql_to_execute.split(';')
+            for i in range(0, len(sql_to_execute)):
+                _run_custom_sql_script(sql_to_execute[i])
+
+    def update_bihar_api_data(self, month):
```

Comment:
for bihar api, table you can directly call `custom.icds_reports.tasks._agg_bihar_api_demographics` with the month as argument.

Reply:
updated!